vizro-core/src/vizro/models/_components/form/dropdown.py

```diff
@@ -39,6 +39,17 @@ def _calculate_option_height(full_options: OptionsType) -> int:
     return 8 + 24 * number_of_lines
 
 
+def _add_select_all_option(full_options: OptionsType) -> OptionsType:
+    """Adds a 'Select All' option to the list of options."""
+    options_dict = [
+        cast(OptionsDictType, {"label": option, "value": option}) if not isinstance(option, dict) else option
+        for option in full_options
+    ]
```

Comment:
Let's add this TODO and then resolve the [conversation](https://github.com/mckinsey/vizro/pull/961#discussion_r1923356781). 
```suggestion
        # TODO: Move option to dictionary conversion within `get_options_and_default` function as here: https://github.com/mckinsey/vizro/pull/961#discussion_r1923356781 
        options_dict = [
        cast(OptionsDictType, {"label": option, "value": option}) if not isinstance(option, dict) else option
        for option in full_options
    ]
```